quil-rs/src/instruction/mod.rs

```diff
@@ -233,26 +240,48 @@ fn write_qubit_parameters(f: &mut fmt::Formatter, qubits: &[Qubit]) -> fmt::Resu
     Ok(())
 }
 
-pub fn get_expression_parameter_string(parameters: &[Expression]) -> String {
+pub fn write_comma_separated_list(
+    f: &mut fmt::Formatter,
+    values: &[impl fmt::Display],
+) -> fmt::Result {
+    if values.is_empty() {
+        return Ok(());
+    }
+
+    write!(f, "{}", values[0])?;
+    for value in values[1..].iter() {
+        write!(f, ", {value}")?;
+    }
+    Ok(())
```

Comment:
This works fine as-is, but stylistically could be written better. My preference is usually

```suggestion
    let mut iter = values.iter();
    
    if let Some(value) = iter.next() {
        write!(f, "{value}")?;
    }
    
    for value in iter {
        write!(f, ", {value}")?;
    }
    
    Ok(())
```

But I'm also okay with

```suggestion
    for (i, value) in values.iter().enumerate() {
        if i != 0 {
            write!(f, ", ")?;
        }
        write!(f, "{value}")?;
    }
    Ok(())
```